client.go

```diff
@@ -62,7 +52,20 @@ type ClientInt interface {
 // information can be found in their respective documentation.
 // Numerous connection options may be specified by configuring a
 // and then supplying a ClientOptions type.
-type Client struct {
+
+type Client interface {
+	IsConnected() bool
+	Connect() Token
+	Disconnect(uint)
+	disconnect()
```

Comment:
A small comment here, having a private method in an interface makes it impossible for external packages to modify the implementation of that method. Which would be beneficial in testing.